### DISCUSSION TITLE
Docker Compose env to secrets

### DISCUSSION THREAD

**Discussion by Technetium1** [docmost/docmost#38] [Ideas]
From https://docs.docker.com/compose/environment-variables/set-environment-variables/
> Don't use environment variables to pass sensitive information, such as passwords, in to your containers. Use [secrets](https://docs.docker.com/compose/use-secrets/) instead.

This would require major reformatting and an update to the instructions, would it be good to do this before a stable release is tagged? Is it worth the effort to do it "correctly" in the first place?

---

`DATABASE_URL` shouldn't need password defined statically here when using env, but **_probably_** needs to be altered if moving to secrets because secrets are instead mounted to `/run/secrets/secretname`
https://github.com/docmost/docmost/blob/a0b31804d987547783435205e54f679af9e3fa4e/docker-compose.yml#L11-L12
https://github.com/docmost/docmost/blob/a0b31804d987547783435205e54f679af9e3fa4e/docker-compose.yml#L25

---

These env entries are also relevant:

https://github.com/docmost/docmost/blob/a0b31804d987547783435205e54f679af9e3fa4e/.env.example#L6
https://github.com/docmost/docmost/blob/a0b31804d987547783435205e54f679af9e3fa4e/.env.example#L17-L21
https://github.com/docmost/docmost/blob/a0b31804d987547783435205e54f679af9e3fa4e/.env.example#L31-L35

**Comment by alohl669** [comment]
I agree with you. In addition to the security issue, It is quite uncommon to see today applications using the password in the URI endpoint

**Comment by aurelien-clu** [comment]
I have not checked your code in details but here is what I have done in the past using `rust`:

```rust
fn load_secret(path: &str, target: &str) {
    match std::fs::read_to_string(path) {
        Ok(secret) => {
            let secret = clean(&secret);
            std::env::set_var(target, secret);
        }
        Err(_) => eprintln!("Secret not found: {path}"),
    }
}

fn clean(secret: &str) -> &str {
    // you may get a trailing line jump, this removes it
    secret.trim().trim_end_matches('\n')
}
```

You could perhaps do something similar.

Note: I don't claim this code to be perfect but it worked for me 😇 


Edit:

Usage example:

```rust
load_secret("/run/secrets/APP_SECRET", "APP_SECRET");
```
